src/MergeContext.js

```diff
@@ -985,6 +987,7 @@ class PullRequest {
 
         if (!(await this._stagedCommitIsFresh())) {
             await this._labels.addImmediately(Config.abandonedStagingChecksLabel());
+            this._abandoned = true;
```

Comment:
Could the label be set (from GitHub point of view) at this point already? Just a question; I am trying to understand the code; not requesting any changes here.